dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,253 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+import os
+import sys
+import json
+import subprocess
+
+from sparktestsupport import SPARK_HOME, ERROR_CODES
+from sparktestsupport.shellutils import exit_from_command_with_retcode, run_cmd, rm_r
+
+
+def print_err(*args):
+    """
+    Given a set of arguments, will print them to the STDERR stream
+    """
+    print(*args, file=sys.stderr)
+
+
+def post_message(mssg, comments_url):
+    http_code_header = "HTTP Response Code: "
+    posted_message = json.dumps({"body": mssg})
+
+    print("Attempting to post to Github...")
+
+    # we don't want to call `run_cmd` here as, in the event of an error, we DO NOT
+    # want to print the GITHUB_OAUTH_KEY into the public Jenkins logs
+    curl_proc = subprocess.Popen(['curl',
+                                  '--silent',
+                                  '--user', 'x-oauth-basic:' + os.environ['GITHUB_OATH_KEY'],
+                                  '--request', 'POST',
+                                  '--data', posted_message,
+                                  '--write-out', http_code_header + '%{http_code}',
+                                  '--header', 'Content-Type: application/json',
+                                  comments_url],
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE)
+    curl_stdout, curl_stderr = curl_proc.communicate()
+    curl_returncode = curl_proc.returncode
+    # find all lines relevant to the Github API response
+    api_response = "\n".join([l for l in curl_stdout.split('\n')
+                              if l and not l.startswith(http_code_header)])
+    # find the line where `http_code_header` exists, split on ':' to get the
+    # HTTP response code, and cast to an int
+    http_code = int(curl_stdout[curl_stdout.find(http_code_header):].split(':')[1])
+
+    if not curl_returncode == 0:
+        print_err("Failed to post message to GitHub.")
+        print_err(" > curl_status:", curl_returncode)
+        print_err(" > curl_output:", curl_stdout)
+        print_err(" > data:", posted_message)
+
+    if http_code and not http_code == 201:
+        print_err(" > http_code:", http_code)
+        print_err(" > api_response:", api_response)
+        print_err(" > data:", posted_message)
+
+    if curl_returncode == 0 and http_code == 201:
+        print(" > Post successful.")
+
+
+def send_archived_logs():
```

Comment:
I think that we should keep this feature for now, since I don't necessarily trust the stability / configurability of the Jenkins plugin that also did this.  Disregard my earlier comments / proposal to remove this.  Let's refactor the code a bit, but we should definitely keep this.